uli_init/tests/test_systems.py

```diff
@@ -1,36 +1,106 @@
-
-    def test_build_pekk(self):
-        for i in range(5):
-            compound = simulate.build_molecule('PEKK', i+1, para_weight=0.50)
-
-
-    def test_para_weight(self):
-        all_para = simulate.random_sequence(para_weight=1, length=10)
-        all_meta = simulate.random_sequence(para_weight=0, length=10)
-        assert all_para.count('para') == 10
-        assert all_meta.count('meta') == 10
-        random.seed(24)
-        mix_sequence = simulate.random_sequence(para_weight=0.50, length=10)
-        assert mix_sequence.count('para') == 4
-        random.seed()
-        mix_sequence = simulate.random_sequence(para_weight = .70, length=100)
-        assert 100 - mix_sequence.count('para') == mix_sequence.count('meta')
-
-    def test_gaff(self):
-        pass
-        
-
-    def test_opls(self):
-        pass
+import pytest
+from uli_init import simulate
+from uli_init.tests.base_test import BaseTest
+import random
+
+class TestSystems(BaseTest):
+
+    def test_build_peek(self):
+        for i in range(5):
+            compound = simulate.build_molecule('PEEK', i+1, para_weight=0.50)
+
+    def test_build_pekk(self):
+        for i in range(5):
+            compound = simulate.build_molecule('PEKK', i+1, para_weight=0.50)
+
+    def test_para_weight(self):
+        all_para = simulate.random_sequence(para_weight=1, length=10)
+        all_meta = simulate.random_sequence(para_weight=0, length=10)
+        assert all_para.count('para') == 10
+        assert all_meta.count('meta') == 10
+        random.seed(24)
+        mix_sequence = simulate.random_sequence(para_weight=0.50, length=10)
+        assert mix_sequence.count('para') == 4
+        random.seed()
+        mix_sequence = simulate.random_sequence(para_weight = .70, length=100)
+        assert 100 - mix_sequence.count('para') == mix_sequence.count('meta')
+
+    # check the base functionality
+    def test_simple_system(self):
+        simple_system = simulate.System(molecule='PEEK', para_weight=0.60,
+                               density=1, n_compounds=[1], polymer_lengths=[2],
+                               forcefield=None, assert_dihedrals=False, remove_hydrogens=False)
+
+    # check that we can load forcefield files
+    def test_load_forcefiled(self):
+        simple_system = simulate.System(molecule='PEEK', para_weight=0.60,
+                               density=1, n_compounds=[1], polymer_lengths=[2],
+                               forcefield='gaff', assert_dihedrals=False, remove_hydrogens=False)
+
+    # check that we can remove H atoms
+    def test_remove_hydrogens(self):
+        simple_system = simulate.System(molecule='PEEK', para_weight=0.60,
+                               density=1, n_compounds=[1], polymer_lengths=[2],
+                               forcefield=None, assert_dihedrals=False, remove_hydrogens=True)
+
+    # test our dihedral checking
+    def test_test_dihedrals(self):

```

Comment:
The `assert_dihedrals` parameter only gets used by foyer when a force field is passed. So, currently this test isn't testing that all dihedrals are typed successfully.